staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/queueset.go

```diff
@@ -674,14 +701,33 @@ func (qs *queueSet) selectQueueLocked() *queue {
 		qs.robinIndex = (qs.robinIndex + 1) % nq
 		queue := qs.queues[qs.robinIndex]
 		if queue.requests.Length() != 0 {
-			currentVirtualFinish := queue.GetNextFinish(qs.estimatedServiceTime)
+			// the virtual finish time of the oldest request is:
+			//   virtual start time + G
+			// We are not taking the width of the request into account when
+			// we calculate the virtual finish time of the request because
+			// it can starve requests with smaller wdith in other queues.
```

Comment:
one last nit: can you maybe add a specific example that we were discussing (I mean your simplified version with two queues) and discussion about it (e.g. need to process 60,000 requests before we put from the other queue) to ensure we won't forget about this case in the future?

Reply:
done, pls check

Reply:
I do not think we should attempt a justification in a comment here.  The original fair queuing paper says that the dispatch order is the order of completion in the virtual world (with ties broken by round-robin ordering), and none of our design alternatives change that.

Reply:
The example below shows that if we didn't change what we had before, it wouldn't work. So I actually claim that this comment is really needed.

I'm happy to hear how we can frame it differently, but I don't think that's very urgent.